samza-kv/src/main/scala/org/apache/samza/storage/kv/AccessLoggedStore.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.storage.kv
+
+import java.util
+
+import org.apache.samza.config.StorageConfig
+import org.apache.samza.task.MessageCollector
+import org.apache.samza.util.Logging
+import org.apache.samza.system.{OutgoingMessageEnvelope, SystemStream, SystemStreamPartition}
+import org.apache.samza.serializers._
+
+class AccessLoggedStore[K, V](
+    val store: KeyValueStore[K, V],
+    val collector: MessageCollector,
+    val systemStreamPartition: SystemStreamPartition,
+    val storageConfig: StorageConfig,
+    val storeName: String) extends KeyValueStore[K, V] with Logging {
+
+  object DBOperations extends Enumeration {
+    type DBOperations = Value
+    val READ = Value("read")
+    val WRITE = Value("write")
+    val DELETE = Value("delete")
+    val RANGE = Value("range")
+    val READ_ALL = Value("read_all")
+    val WRITE_ALL = Value("write_all")
+    val DELETE_ALL = Value("delete_all")
+  }
+
+  val streamName = storageConfig.getAccessLogStream(systemStreamPartition.getSystemStream.getStream)
+  val systemStream = new SystemStream(systemStreamPartition.getSystemStream.getSystem, streamName)
+  val partitionId: Int = systemStreamPartition.getPartition.getPartitionId
+  val serializer = new StringSerde("UTF-8")
+  val sample = storageConfig.getSamplingSetting(storeName)
+  val generator = scala.util.Random
+  val DELIMITER = " || "
+
+  def get(key: K): V = {
+    var message = DBOperations.READ + DELIMITER + key
```

Comment:
Issue: The "+ key" is going to call key.toString, which is likely not what you want. Same for the toStringKey method.
I'd recommend using binary format for the message and putting the serialized key as a binary blob in it.